lib/rule-tester/rule-tester.js

```diff
@@ -1173,6 +1191,10 @@ class RuleTester {
 								error.message,
 							);
 						} else if (hasOwnProperty(error, "messageId")) {
+							assert.ok(
+								requireMessage !== "message",
+								"The test run requires 'message' but the error (also) uses 'messageId'.",
```

Comment:
```suggestion
								"The test run requires 'message' but the error uses 'messageId'.",
```

At this point we certainly know that `message` property is not specified.